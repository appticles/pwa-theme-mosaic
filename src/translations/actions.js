import config from 'react-global-configuration';

export const REQUEST_TRANSLATIONS = 'REQUEST_TRANSLATIONS';
export const RECEIVE_TRANSLATIONS = 'RECEIVE_TRANSLATIONS';

export const requestTranslations = () => ({
  type: REQUEST_TRANSLATIONS,
});

export const receiveTranslations = translations => ({
  type: RECEIVE_TRANSLATIONS,
  translations,
});

export const fetchTranslations = (dispatch) => {
  dispatch(requestTranslations());
  config.set(window.__INITIAL_CONFIG__, { freeze: false });

  return fetch(config.get('translate').path, {
    method: 'GET',
    header: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      mode: 'cors',
    },
  })
    .then(response => response.json())
    .then(json => dispatch(receiveTranslations(json)))
    .catch(() => {
      dispatch(receiveTranslations([]));
    });
};
