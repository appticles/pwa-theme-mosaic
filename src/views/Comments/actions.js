import config from 'react-global-configuration';

export const REQUEST_COMMENTS = 'REQUEST_COMMENTS';
export const RECEIVE_COMMENTS = 'RECEIVE_COMMENTS';

export const requestComments = () => ({
  type: REQUEST_COMMENTS,
});

export const receiveComments = comments => ({
  type: RECEIVE_COMMENTS,
  comments,
});

export const fetchComments = (params = {}) => (dispatch) => {
  dispatch(requestComments());

  let url;
  if (params) {
    url =
      config.get('export').comments +
      '?' +
      Object.keys(params)
        .map(k => k + '=' + encodeURIComponent(params[k]))
        .join('&');
  }
  return fetch(url)
    .then(response => response.json())
    .then(json => dispatch(receiveComments(json)))
    .catch(() => {
      dispatch(receiveComments([]));
    });
};

export const addComment = (params = {}) => {
  fetch(config.get('export').comments, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ params }),
  });
};
