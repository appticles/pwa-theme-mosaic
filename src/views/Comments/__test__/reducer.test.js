import comments, { INITIAL_STATE, getComments, getCommentsFetching } from '../reducers';
import { requestComments, receiveComments } from '../actions';

describe('Comments reducer', () => {
  it('returns same state on an unhandled action', () => {
    expect(comments(INITIAL_STATE, { type: '_NULL' })).toMatchSnapshot();
  });
  it('handles REQUEST_COMMENTS action', () => {
    const commentsList = {
      isFetching: 0,
      items: [
        {
          id: 43,
          post: 63,
          author_name: 'Some Author',
          content: {
            rendered: 'Some comment content',
          },
          date: 'some-date',
        },
        {
          id: 43,
          post: 63,
          author_name: 'Some Author',
          content: {
            rendered: 'Some comment content',
          },
          date: 'some-date',
        },
      ],
    };
    expect(comments(commentsList, requestComments())).toMatchSnapshot();
  });
  it('handles RECEIVE_COMMENTS action case 1', () => {
    const postState = {
      isFetching: 1,
      items: [
        {
          id: 43,
          post: 63,
          author_name: 'Some Author',
          content: {
            rendered: 'Some comment content',
          },
          date: 'some-date',
        },
        {
          id: 43,
          post: 63,
          author_name: 'Some Author',
          content: {
            rendered: 'Some comment content',
          },
          date: 'some-date',
        },
      ],
    };
    const commentsList = [
      {
        id: 43,
        post: 63,
        author_name: 'Some Author',
        content: {
          rendered: 'Some comment content',
        },
        date: 'some-date',
      },
    ];
    expect(comments(postState, receiveComments(commentsList))).toMatchSnapshot();
  });
  it('handles RECEIVE_COMMENTS action case 2', () => {
    const postState = {
      isFetching: 1,
      items: [
        {
          id: 43,
          post: 63,
          author_name: 'Some Author',
          content: {
            rendered: 'Some comment content',
          },
          date: 'some-date',
        },
        {
          id: 43,
          post: 63,
          author_name: 'Some Author',
          content: {
            rendered: 'Some comment content',
          },
          date: 'some-date',
        },
      ],
    };
    const commentsList = {
      id: 43,
      post: 63,
      author_name: 'Some Author',
      content: {
        rendered: 'Some comment content',
      },
      date: 'some-date',
    };
    expect(comments(postState, receiveComments(commentsList))).toMatchSnapshot();
  });

  // reducer functions
  it('getCommentsFetching returns state.isFetching', () => {
    expect(getCommentsFetching(INITIAL_STATE)).toBe(0);
  });
  it('getComments returns state.items', () => {
    const state = {
      items: [
        {
          id: 43,
          post: 63,
          author_name: 'Some Author',
          content: {
            rendered: 'Some comment content',
          },
          date: 'some-date',
        },
      ],
    };
    expect(getComments(state)).toBe(state.items);
  });
});
