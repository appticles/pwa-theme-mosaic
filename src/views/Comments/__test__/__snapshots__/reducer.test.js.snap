// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments reducer handles RECEIVE_COMMENTS action case 1 1`] = `
Object {
  "isFetching": 0,
  "items": Array [
    Object {
      "author_name": "Some Author",
      "content": Object {
        "rendered": "Some comment content",
      },
      "date": "some-date",
      "id": 43,
      "post": 63,
    },
  ],
}
`;

exports[`Comments reducer handles RECEIVE_COMMENTS action case 2 1`] = `
Object {
  "isFetching": 0,
  "items": Array [
    Object {
      "author_name": "Some Author",
      "content": Object {
        "rendered": "Some comment content",
      },
      "date": "some-date",
      "id": 43,
      "post": 63,
    },
  ],
}
`;

exports[`Comments reducer handles REQUEST_COMMENTS action 1`] = `
Object {
  "isFetching": 1,
  "items": Array [
    Object {
      "author_name": "Some Author",
      "content": Object {
        "rendered": "Some comment content",
      },
      "date": "some-date",
      "id": 43,
      "post": 63,
    },
    Object {
      "author_name": "Some Author",
      "content": Object {
        "rendered": "Some comment content",
      },
      "date": "some-date",
      "id": 43,
      "post": 63,
    },
  ],
}
`;

exports[`Comments reducer returns same state on an unhandled action 1`] = `
Object {
  "isFetching": 0,
  "items": Array [],
}
`;
